include/dlaf/permutations/general/impl.h

```diff
@@ -152,4 +157,345 @@ void Permutations<B, D, T, C>::call(SizeType i_begin, SizeType i_end, Matrix<con
                                                false>(dlaf::internal::Policy<B>(), std::move(permute_fn),
                                                       std::move(sender)));
 }
+
+template <class T, Device D>
+auto whenAllReadWriteTilesArray(LocalTileIndex begin, LocalTileIndex end, Matrix<T, D>& matrix) {
+  LocalTileSize sz{end.row() - begin.row() + 1, end.col() - begin.col() + 1};
+  namespace ex = pika::execution::experimental;
+  namespace ut = matrix::util;
+  return ex::when_all_vector(ut::collectReadWriteTiles(begin, sz, matrix));
+}
+
+template <class T, Device D>
+auto whenAllReadWriteTilesArray(Matrix<T, D>& matrix) {
+  namespace ex = pika::execution::experimental;
+  namespace ut = matrix::util;
+  return ex::when_all_vector(
+      ut::collectReadWriteTiles(LocalTileIndex(0, 0), matrix.distribution().localNrTiles(), matrix));
+}
+
+template <class T, Device D>
+auto whenAllReadOnlyTilesArray(LocalTileIndex begin, LocalTileIndex end, Matrix<const T, D>& matrix) {
+  LocalTileSize sz{end.row() - begin.row() + 1, end.col() - begin.col() + 1};
+  namespace ex = pika::execution::experimental;
+  namespace ut = matrix::util;
+  return ex::when_all_vector(ut::collectReadTiles(begin, sz, matrix));
+}
+
+template <class T, Device D>
+auto whenAllReadOnlyTilesArray(Matrix<const T, D>& matrix) {
+  namespace ex = pika::execution::experimental;
+  namespace ut = matrix::util;
+  return ex::when_all_vector(
+      ut::collectReadTiles(LocalTileIndex(0, 0), matrix.distribution().localNrTiles(), matrix));
+}
```

Comment:
Yes, I support the effort to try to unify the two functions. Moreover, we should consider if we prefer to return a sender with `ex::when_all_vector()` instead of a `std::vector<>` as what the 4 `whenAllRead... ()` overloads are doing.

Reply:
Fully agree!